### ISSUE TITLE
Hystrix without Archaius

### ISSUE THREAD

**Issue by daijithegeek** [Netflix/Hystrix#129]
Hi,

I tried to implement a custom HystrixPropertiesStrategy, based on our company configuration mechanism. I was wondering if it is possible to get rid of the Archaius dependency as I really do not want to use it.

Indeed, any HystrixCommandProperties implementation need to call the super() constructor which enables Archaius class access.

Another option is to implement an configuration datasource for Archaius based on our configuration mechanism. I'm not a big fan of it for different reasons. The main one is the fact that I am currently developing a library, and archaius comes with guava and commons-everything dependencies. Therefore I will force library client to get those dependencies, adding many potential conflict on their side.

Am I missing something important there ? Is there a better way?

Thank you!
David

**Comment by agentgt** [comment]
I would not have a problem with the Archaius dependency if Archaius itself did not depend on so many things that have static initialization (most notably SLF4J and commons-logging).

There really needs to be a boot strap configuration library that does not depend on a logger or various other infrastructure as you get a nice chicken and egg problem of who does the initial configuration. For example you cannot configure your logging infrastructure with Archaius because it will already have instantiated logging. While you can possibly just live with the fact that your logging infrastructure will have to reinitialize this has historically been problematic with libraries such as logback (in fact there was no way to control the initialization of logback till [I filed a bug](https://github.com/qos-ch/logback/pull/181)).

One of the ways to mitigate dispatching to an implementation with out relying on system properties or environment variables is to use the JDK Service Loader (which is what I did for logback). Consequently depending on what jars are in the classpath the right plugins would then be loaded automatically.

TLDR ... depending on libraries that have any sort of dependency of static initialization is hard to remove and makes it difficult for those who care about initialization (usually logging is the primary issue).

**Comment by mattrjacobs** [comment]
@agentgt I agree 100%.  I'll be working on this soon

**Comment by agentgt** [comment]
For future folks that land here see #970
